server/loops.lua

```diff
@@ -1,16 +1,18 @@
 local config = require 'config.server'
 
 local function removeHungerAndThirst(src, player)
-    local newHunger = player.PlayerData.metadata.hunger - config.player.hungerRate
-    local newThirst = player.PlayerData.metadata.thirst - config.player.thirstRate
+    local playerState = Player(src).state
+    if not playerState.isLoggedIn then return end
+    local newHunger = playerState.hunger - config.player.hungerRate
+    local newThirst = playerState.thirst - config.player.thirstRate
     if newHunger <= 0 then
         newHunger = 0
     end
     if newThirst <= 0 then
         newThirst = 0
     end
-    player.Functions.SetMetaData('thirst', newThirst)
-    player.Functions.SetMetaData('hunger', newHunger)
```

Comment:
Don't we need to continue to keep the metadata updated?

Reply:
It's saved anyway,
if you log out / drop it's saved,
when it's updated via the loop it's saved.

So i think it's safe to not use SetMetadata everytime.

Reply:
I would test this so we know

Reply:
We need to continue to trigger the metadata set event when metadata is changed to avoid breaking our API contract https://docs.qbox.re/resources/qbx_core/events/server#qbx_coreserveronsetmetadata